### ISSUE TITLE
Record the last modification date for the custom stats tables

### ISSUE THREAD

**Issue by elizoller** [asulibraries/islandora-repo#359]
It would be handy to have the last modification date/time stored in the asu_collection_extras_item_downloads and asu_collection_extras_collection_usage tables so we can verify when/if certain rows have been updated

**Comment by wgilling** [comment]
Adjusted the code in the install schema and added a hook_update_8001 that will be triggered by running update.php.

Instead of making the field defined as an auto-update, the insert statements were adjusted to set the timestamp.

**Comment by wgilling** [comment]
We should try to query Matomo in a more strategic way --- if it is possible to query Matomo for "what has changed since the last request??" and if that is able to return a set of items, then ONLY need to process those items.

If that doesn't work, we should certainly try to dynamically process multiple items in a single call to Matomo - to process however many items at a time... get the results for those number of items ($howmany parameter), parse those results, and build up the next set of however many, parse them, repeat until the CRON has processed the configured number of items.

**Comment by wgilling** [comment]
I looked at several different Matomo API calls - https://developer.matomo.org/api-reference/reporting-api but 

Something like this may work for the page views: https://keep.lib.asu.edu/matomo/index.php?module=API&method=Actions.getPageUrls&idSite=1&filter_limit=-1&expanded=1&period=range&date=2020-01-01,2021-04-13&format=json&token_auth=

The strategy that I just spoke with Eli about was a 2 part refactoring... Since we really do not want this process to calculate the views/downloads for the entire repository EVERY day, the first part is to populate the initial set of records. The second refactoring would be to add a system that will fetch the matomo numbers for a given item EVERY time it is viewed.

In order to calculate the initial set of values for all items, this must be broken down into smaller chunks. The tables could be populated with each node's id, the collection id, and a NULL modified date. A CRON process could process a given number of those "NULL modified date" records in each iteration.

**Comment by wgilling** [comment]
After the refactoring of how the views and downloads are populated, the following data is in my database tables:
```
mysql> select * from ace_relations;
+-----+------------+-----------+-------------+
| uid | parent_nid | child_nid | parent_type |
+-----+------------+-----------+-------------+
|   4 |       5626 |      5623 | collec      |
|   5 |         49 |      5623 | item        |
|   6 |       5744 |      5623 | item        |
|   7 |       5744 |      5626 | collec      |
|   8 |       5744 |      5745 | collec      |
|   9 |         51 |      5812 | collec      |
+-----+------------+-----------+-------------+
6 rows in set (0.00 sec)

mysql> select * from ace_items;
+-------+-------+-----------+----------------+--------------------+
| i_nid | views | downloads | views_modified | downloads_modified |
+-------+-------+-----------+----------------+--------------------+
|  5623 |     2 |         0 |     1619541197 |         1619541197 |
|  5626 |     1 |         0 |     1619541198 |         1619541198 |
|  5740 |     1 |         0 |     1619541198 |         1619541198 |
|  5742 |     1 |         0 |     1619541198 |         1619541198 |
|  5745 |     8 |         0 |     1619541198 |         1619541198 |
|  5812 |     2 |         3 |     1619541199 |         1619541199 |
+-------+-------+-----------+----------------+--------------------+
6 rows in set (0.00 sec)

```

**Comment by wgilling** [comment]
adjust the collection stats box display to add a separate box for the downloads value. Insert this box immediately after the views box.

**Comment by elizoller** [comment]
closed with #382
